src/tarball.py

```diff
@@ -89,12 +89,16 @@ def _push_tarball(self, config, describe):
         os.unlink(tarball_path)
         return tarball
 
-    def _update_node(self, node, describe, tarball, status):
-        node['revision']['describe'] = describe
-        node['status'] = status
-        node['artifacts'] = {
-            'tarball': urllib.parse.urljoin(self._storage_url, tarball),
+    def _create_node(self, checkout_node, describe, tarball, status):
+        node = {
+            'parent': checkout_node['_id'],
+            'name': 'create-tarball',
+            'artifacts': {'tarball': urllib.parse.urljoin(self._storage_url,
+                                                          tarball)},
+            'revision': checkout_node['revision'],
+            'status': status,
         }
+        node['revision']['describe'] = describe
```

Comment:
Ah, I think that's a separate thing. The `describe` value isn't known when the `checkout` node is first created. So here we should be updating it in the `checkout` node like before, and then also create a `tarball` child node.